### ISSUE TITLE
Even with Shields Down, I need to go into Settings -> Privacy to disable blocking 3rd party cookies.

### ISSUE THREAD

**Issue by butlermatt** [brave/browser-laptop#2417]
**Did you search for similar issues before submitting this one?**
Yes
**Describe the issue you encountered:**
A product of the company I use, uses localStorage. While "Block 3rd Party Cookies" is enabled, I get the following error in the JS Console:

> Uncaught SecurityError: Failed to read the 'localStorage' property from 'Window': Access is denied for this document.

Even if I enable "Shields down" for that site, I still continue to receive that error. I need to go into Settings, 
Privacy and Allow All Cookies, before I am able to log into the product (before the product can access LocalStorage).

**Expected behavior:**
To log in. (That when shields are down, at the least, that 3rd party cookies for connections on that tab are enabled at minimum). Ideally having shields up wouldn't cause the issue at all. It appears the "3rd party requests" are actually coming from a blob URL from the actual site I'm loading itself anyways.
- Platform (Win7, 8, 10? macOS? Linux distro?):
  Mac OS X 10.11.5 (64-bit)
- Brave Version:
  Brave: 0.11.0
  Electron: 1.2.3
  libchromiumcontent: 51.0.2704.103
  V8: 5.1.281.65
  Node.js: 6.1.0
  Update Channel: dev
- Steps to reproduce:
  Unfortunately i don't have access to provide on the site that can't complete the login due to this error. Though happy to try and provide any assistance in debugging.
- Screenshot if needed:
- Any related issues:

**Comment by butlermatt** [comment]
For the record, I can leave block 3rd party cookies enabled in Chrome and still be able to log into our App

**Comment by diracdeltas** [comment]
Does https://jsfiddle.net/7ke9r14a/7/ work as expected with shields down? You should see an incrementing strings of 0's in the `local storage:` output if so.

It works for me on brave 0.11 os x

**Comment by butlermatt** [comment]
Yes, I see _a_ string with a 0 in the local storage output (not multiple). Full output is here:

```
local storage:
"0"
session storage:
"0"
indexeddb:
{"0":"idb"}
cookies:
"abc=123"
websql:
{"0":{"tbl_name":"__WebKitDatabaseInfoTable__"},"1":{"tbl_name":"todo"}}
filesystem API:
"https_fiddle.jshell.net_0:Persistent"
```

**Comment by diracdeltas** [comment]
local storage unblock is working in that case; may be a referer issue

**Comment by tivac** [comment]
I see this when trying to access my local [Sonarr](https://sonarr.tv/) instance.

- Hitting "Shields Down" doesn't fix the issue.
- Changing 3rd party cookie setting in Preferences -> Security to "All" does fix it.
- Fiddle output matches what @butlermatt posted.

**Comment by molinadavid** [comment]
I can confirm the same issue, I am on Mac OS X 10.11.6

Brave: 0.11.5
Electron: 1.3.3
libchromiumcontent: 52.0.2743.116
V8: 5.2.361.49
Node.js: 6.3.0
Update Channel: dev

I am building a Phonegap application that uses local storage but I get the same error than @tivac  and @butlermatt even when the shield are down for that page, the only way to make it work is by allowing 3rd party cookies on the global settings.

**Comment by zhaozhiming** [comment]
same bug to me

**Comment by diracdeltas** [comment]
I still can't repro this. Anyone have a specific site / test case?

If the fiddle output in https://jsfiddle.net/7ke9r14a/7/ is non-empty when shields are down, then it is working as expected

**Comment by zhaozhiming** [comment]
@diracdeltas I use your jsfiddle and the console print error : 
![image](https://cloud.githubusercontent.com/assets/1436694/19216973/b35f69fe-8e04-11e6-945f-2388f264a129.png)

**Comment by dangerginger** [comment]
I am getting this error with the neo4j browser connection.  I run neo4j on the local network and connect to the standard port at `http://10.1.1.123:7474`.  I get the error below, which is the same as mentioned in this issue.  

* Turning shields off does not resolve the problem.  After disabling shields for that url, the cookie policy dialog says `Allow All Cookies`.  However, I get the same error.  

* Going into settings and changing the `Cookie Control` policy under `Default Shields: All Sites` to `Allow All Cookies` causes the site to work properly.

* I don't understand why the default shields and URL-specific shields cookie control policies have different effects.  I would like to use the neo4j browser which I trust, but continue blocking third-party cookies by default.

* Using chromium, I can set the option `block third-party cookies and site data`.  The neo4j browser works as expected either way.

![2017-01-12-074608_879x61_scrot](https://cloud.githubusercontent.com/assets/14853111/21896460/4dbb2282-d89b-11e6-9036-ec080674c61a.png)

```
Name	Version
Brave	0.12.15
Muon	1.4.31
libchromiumcontent	53.0.2785.143
V8	5.3.332.47
Node.js	6.5.0
Update Channel	dev
os.platform	linux
os.release	4.8.13-1-ARCH
os.arch	x64
```

**Comment by hmrodrigues** [comment]
I have the same problem when connecting to Odoo on Ubuntu Trusty

Brave	0.12.15
Muon	1.4.31
libchromiumcontent	53.0.2785.143
V8	5.3.332.47
Node.js	6.5.0
Update Channel	dev
os.platform	linux
os.release	3.13.0-106-generic
os.arch	x64

**Comment by superseb** [comment]
Got the same error when running Rancher.

To reproduce (run this on a remote server, localhost works fine): 
`docker run -d -p 8080:8080 rancher/server:v1.3.4`

Open remote_ip:8080

```
Brave: 0.13.2 
rev: 25b1199fb6154b089cbad37926483239495b9800 
Muon: 2.0.19 
libchromiumcontent: 54.0.2840.100 
V8: 5.4.500.41 
Node.js: 7.0.0 
Update Channel: dev 
os.platform: darwin 
os.release: 16.4.0 
os.arch: x64
```

**Comment by grantwinney** [comment]
Same problem here:

> Uncaught DOMException: Failed to read the 'localStorage' property from 'Window': Access is denied for this document.

The site is running using Amazon AWS and is only accessible from my company's IP address. It was working fine in Brave all last week, but suddenly today (with no changes to the site itself) Brave wouldn't load it. It loaded fine in Chrome and Firefox.

I cleared cache and cookies, then cleared everything and restarted the browser. Then I disabled shields on the site. Nothing.

Finally came across this issue, changed the cookies setting to allow all cookies, and it loaded just fine.

Brave: 0.13.2
rev: 25b1199
Mac OS X 10.11.6 El Capitan
